### PR TITLE
Feature/open 82

### PR DESCRIPTION
**Pull request by delrueba**

### CODE REVIEW

File: tests/controllers/UI/ChannelControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    {
+        $this->requestsByUserWithRoleAndCheckStatusCode($userRole, $verb, $pathArg, $data, $statusCode);
+    }
+
+    /**
+     * assemble the path on the given params
+     */
+    protected function assemblePath($params)
+    {
+        return $this->apiUrl . '/' . $params;
```
Comment (MPParsley) [inline tests/controllers/UI/ChannelControllerTest.php]: Are these params filtered?

File: tests/TestCase.php [hunk 2]
```diff
@@ -108,4 +115,16 @@ public function getContentStructureTested()
     protected function extraStructureTest($content)
     {
     }
+
+    public function requestsByUserWithRoleAndCheckStatusCode($userRole, $verb, $pathArg, $data, $statusCode)
+    {
+        if ($userRole !== 'unauth') {
```
Comment (MPParsley) [inline tests/TestCase.php]: Use a class constant for 'unauth'.

File: resources/assets/js/mixins/users.js [hunk 3]
```diff
@@ -1,95 +1,107 @@
...
-                    this.fetchUsers();
-                    this.fetchServices();
-                    this.modalClose();
+            // backend will create the user if not found
+            this.$http.post('/api/ui/inviteuser', newRole)
+                .then(({data}) => {
+
+                    // add user to service users
+                    let serviceIndex = this.services.findIndex(s => s.id === newRole.service_id);
+                    if(serviceIndex > -1 && this.services[serviceIndex].users) {
```
Comment (MPParsley) [inline resources/assets/js/mixins/users.js]: if (

File: resources/assets/js/modal/ModalText.vue [hunk 4]
```diff
@@ -194,16 +194,25 @@ export default {
       this.modalWait();
 
       this.modal.strict = true;
+
+      if(!this.modal.usr && !this.validEmail) {
```
Comment (MPParsley) [inline resources/assets/js/modal/ModalText.vue]: if (

File: tests/controllers/UI/ChannelControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    /**
+     * @var string
+     */
+    protected $apiUrl = '/api/ui/services/1/channels';
+    /**
+     * @test
+     */
+    public function testGetNoUiServicesWithoutAuth()
+    {
+        //$this->withoutMiddleware();
```
Comment (MPParsley) [inline tests/controllers/UI/ChannelControllerTest.php]: Remove commented code.

File: tests/TestCase.php [hunk 6]
```diff
@@ -46,29 +44,37 @@ protected function assemblePath($params)
...
             'json' => 'application/json',
             'json-ld' => 'application/ld+json',
             'html' => 'text/html',
             'text' => 'text/plain',
         ];
+
+        $request = 'json';
+        if (isset($data['format'])) {
+            $request = $data['format'];
+            if (!array_key_exists($data['format'], $formats)) {
```
Comment (MPParsley) [inline tests/TestCase.php]: $request = 'json';
        if (isset($data['format']) && array_key_exists($data['format'], $formats)) {
            $request = $data['format'];
        }

File: resources/assets/js/page/PageService.vue [hunk 7]
```diff
@@ -23,12 +22,14 @@
           <tr>
             <th-sort by="name">Naam gebruiker</th-sort>
             <th-sort by="email">E-mailadres</th-sort>
-            <th>Gebruikers beheren</th>
+            <th>Lid of eigenaar</th>
```
Comment (MPParsley) [inline resources/assets/js/page/PageService.vue]: No translations?

File: resources/assets/js/modal/ModalText.vue [hunk 8]
```diff
@@ -194,16 +194,25 @@ export default {
...
+          this.modalResume();
+          return alert('Dit is geen geldig e-mail adres');
```
Comment (MPParsley) [inline resources/assets/js/modal/ModalText.vue]: No translations?

File: resources/assets/js/mixins/users.js [hunk 9]
```diff
@@ -1,95 +1,107 @@
...
                     })
                     .then(this.statusReset)
                     .catch(fetchError)
             }
         },
         translateRole(role) {
-
-            let translation;
-
             switch (role) {
```
Comment (MPParsley) [inline resources/assets/js/mixins/users.js]: Use "constants"?
import {ADMIN, OWNER, MEMBER} from "../constants";

File: resources/assets/js/mixins/modal.js [hunk 10]
```diff
@@ -68,16 +71,16 @@ export default {
     },
 
     newUser(srv) {
-      this.modal.text = 'newUser'
+      this.modal.text = 'newUser';
       this.modal.role = 'Member'
```
Comment (MPParsley) [inline resources/assets/js/mixins/modal.js]: ;

File: resources/assets/js/mixins/modal.js [hunk 11]
```diff
@@ -68,16 +71,16 @@ export default {
...
-      this.modal.srv = srv
+      this.modal.text = 'newRole';
+      this.modal.srv = srv;
     },
     newRoleForUser(usr) {
-      this.modal.text = 'newRoleForUser'
-      this.modal.usr = usr
+      this.modal.text = 'newRoleForUser';
+      this.modal.usr = usr;
       this.modal.role = 'Member'
```
Comment (MPParsley) [inline resources/assets/js/mixins/modal.js]: ;

File: resources/assets/js/mixins/auth.js [hunk 12]
```diff
@@ -55,9 +53,9 @@ export const rootAuthMixin = {
     user: {
       deep: true,
       handler  (v) {
-        console.log('Update user', inert(this.user))
+        console.log('Update user', inert(this.user));
```
Comment (MPParsley) [inline resources/assets/js/mixins/auth.js]: Is debug allowed?

File: resources/assets/js/lib.js [hunk 13]
```diff
@@ -232,7 +232,7 @@ export function fetchError (response) {
     alert(response.body.message);
   }
 
-  console.warn(response);
+  console.error(response);
```
Comment (MPParsley) [inline resources/assets/js/lib.js]: Is debug allowed?

File: app/Http/Controllers/UI/RolesController.php [hunk 14]
```diff
@@ -42,13 +41,33 @@ public function store(StoreRoleRequest $request)
...
+    public function update(StoreRoleRequest $request) {
+        $input = $request->input();
+
+        $success = $this->users->linkToService($input['user_id'], $input['service_id'], $input['role']);
+
+        if ($success) {
+            return response()->json(['role' => $input['role']]);
+        }
+
+        return response()->json(['message' => 'Er is iets misgegaan tijdens het aanpassen van de rol.'], 400);
```
[the remark this answers is not in the document]
Comment (delrueba) [reply app/Http/Controllers/UI/RolesController.php]: Ik vermoed dat Merlijn dit bewust deed. 
Alles onder 'UI' is enkel voor de invoermodule beschikbaar. En die is enkel Nederlandstalig.

File: resources/assets/js/lib.js (hunk 13 above)
Comment (delrueba) [reply resources/assets/js/lib.js]: For the moment, yes.

**Pull request merged**